mmdet/models/detectors/yolact.py

```diff
@@ -0,0 +1,149 @@
+import torch
+
+from mmdet.core import bbox2result, multi_apply
+from ..builder import DETECTORS, build_head
+from .single_stage import SingleStageDetector
+
+
+@DETECTORS.register_module()
+class Yolact(SingleStageDetector):
+    """Implementation of `YOLACT <https://arxiv.org/abs/1904.02689>`_"""
+
+    def __init__(self,
+                 backbone,
+                 neck,
+                 bbox_head,
+                 segm_head,
+                 mask_head,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+        super(Yolact, self).__init__(backbone, neck, bbox_head, train_cfg,
+                                     test_cfg, pretrained)
+        self.segm_head = build_head(segm_head)
+        self.mask_head = build_head(mask_head)
+        self.init_segm_mask_weights()
+        if train_cfg is not None:
+            self.min_gt_box_wh = train_cfg.min_gt_box_wh
+
+    def init_segm_mask_weights(self):
+        """Initialize weights of the YOLACT semg head and YOLACT mask head."""
+        self.segm_head.init_weights()
+        self.mask_head.init_weights()
+
+    def process_gt_single(self, gt_bboxes, gt_labels, gt_masks, min_gt_box_wh,
+                          device):
+        """Cuda the gt_masks and discard boxes that are smaller than we'd
+        like."""
+        gt_masks = torch.from_numpy(gt_masks.masks).to(device)
+        w = gt_bboxes[:, 2] - gt_bboxes[:, 0]
+        h = gt_bboxes[:, 3] - gt_bboxes[:, 1]
+        keep = (w > min_gt_box_wh[0]) * (h > min_gt_box_wh[1])
+        gt_bboxes = gt_bboxes[keep]
+        gt_labels = gt_labels[keep]
+        gt_masks = gt_masks[keep]
+        return gt_bboxes, gt_labels, gt_masks
```

Comment:
Shouldn't these samples be filtered in the dataset?

Reply:
Thanks for your comments. I've finished the code modification except for this one. Correct me if I'm wrong but I don't think mmdet provides the annotation-level sample filter. Are you suggesting to let me implement such a function? If so, which file shall I add to?  Thanks.